### PR TITLE
feat: split view for workflow

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

What I added:
- A panel that will be used to view an asset/process from a workflow node - this has been added into `tera-project.vue`
- I have also made a component called `tera-project-page.vue` to house the possible asset/process components that are viewed in a panel in the project view

How it works/how you can add to this:
- `tera-workflow-node.vue` is where I pass the node to be saved in pinia, I use`openedWorkflowNodeStore` (if you have a better name let me know)
- In the store I save the node as well as its `assetId` and `pageType` for easy access
- I get that node in `tera-project.vue` and pass its `assetId` attribute into `tera-project-page.vue` to view a model asset
- So far, whatever the last output is of your model node will be displayed in the drilldown panel
- If you need to add more attributes from the node or even the node itself feel free to pass it into `tera-project-page.vue` as new prop(s) according to your needs
- Also add your components for new processes such as calibrate into `tera-project-page.vue`

https://github.com/DARPA-ASKEM/Terarium/assets/28237258/4e5b033e-bcb1-47b5-9619-ba4b618a82d8

### CODE REVIEW

File: packages/client/hmi-client/src/components/widgets/tera-infinite-canvas.vue
Comment (shawnyama) [inline packages/client/hmi-client/src/components/widgets/tera-infinite-canvas.vue:1]: Why I got rid of these: 

This 100%s were shifting the whole project component up when opening the new panel. They are redundant anyways since the width and height of the svg is assigned in the HTML

File: packages/client/hmi-client/src/components/widgets/tera-infinite-canvas.vue [hunk 1]
```diff
@@ -164,8 +164,6 @@ main > * {
 
 .background-layer {
 	cursor: grab;
-	width: 100%;
-	height: 100%;
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/widgets/tera-infinite-canvas.vue:168]: Why I got rid of these:

The 100%s were shifting the whole project component up when opening the new panel. They are redundant anyways since the width and height of the svg is assigned in the HTML

**Comment by YohannParis** [comment]
You need to add tabs in that panel, exactly like in tera-project, kind of a tera-project within a tera-project (inception much).
Tell me if that makes sense?

File: packages/client/hmi-client/src/page/project/tera-project.vue [hunk 2]
```diff
@@ -276,21 +229,29 @@ const props = defineProps<{
...
 
 const tabStore = useTabStore();
-const router = useRouter();
-const resources = useResourcesStore();
 
-const newModelId = ref<string>('');
-const isNewModel = ref<boolean>(true);
+// Makes asset from workflow node appear
+const openedWorkflowNodeStore = useOpenedWorkflowNodeStore();
+const workflowNodeAssetId = computed(() => {
```
Comment (echl) [inline packages/client/hmi-client/src/page/project/tera-project.vue:235]: Can you add some comments to explain what is happening here? I find it a bit hard to follow.

File: packages/client/hmi-client/src/page/project/tera-project.vue [hunk 3]
```diff
@@ -276,21 +229,29 @@ const props = defineProps<{
...
+	let index: number = 0;
+
+	if (openedWorkflowNodeStore?.workflowNode?.outputs.length) {
+		index = openedWorkflowNodeStore.workflowNode.outputs.length - 2;
```
Comment (echl) [inline packages/client/hmi-client/src/page/project/tera-project.vue:239]: Why `-2`? If length is 1, won't index be -1?

File: packages/client/hmi-client/src/page/project/components/tera-tab-content.vue [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+			:project="project"
+			@asset-loaded="emit('asset-loaded')"
+		/>
+	</template>
+	<code-editor
+		v-else-if="assetType === ProjectAssetTypes.CODE"
+		:initial-code="code"
+		@on-model-created="openNewModelFromCode"
+	/>
+	<tera-model
```
Comment (echl) [inline packages/client/hmi-client/src/page/project/components/tera-tab-content.vue:45]: Why are there two `tera-models`? Is one for an existing model and one for a new one?

Comment (shawnyama) [reply packages/client/hmi-client/src/page/project/components/tera-tab-content.vue:45]: Yeah the one on the outside is for when you're creating one.

File: packages/client/hmi-client/src/page/project/tera-project.vue (hunk 3 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/page/project/tera-project.vue:239]: Once the model is ran a new output port is added. That new port is empty though so I am just grabbing the one before that. I'll write this better though as I haven't accounted for one empty output port.

File: packages/client/hmi-client/src/page/project/components/tera-tab-content.vue (hunk 4 above)
Comment (echl) [reply packages/client/hmi-client/src/page/project/components/tera-tab-content.vue:45]: Just wondering if there is any value to consolidating them into the same component. Intuitively I would think `tera-model` would display either a new model or saved model instead needing two instances of `tera-model`.

**Comment by echl** [comment]
Is there a way currently to close the split view?

**Comment by shawnyama** [comment]
> Is there a way currently to close the split view?

No, I thought once tabs are implemented you'd close the last tab and that would close the view. At the moment you can just drag it out of sight

File: packages/client/hmi-client/src/page/project/tera-project.vue [hunk 5]
```diff
@@ -276,21 +229,31 @@ const props = defineProps<{
...
+
+// Grabs the model id from the last model config output from the node
+const workflowNodeAssetId = computed(() => {
+	let index: number = 0;
+
+	if (openedWorkflowNodeStore?.workflowNode?.outputs.length) {
+		index = openedWorkflowNodeStore.workflowNode.outputs.length - 1; // Grab index of last output port
+	}
+	return (
+		openedWorkflowNodeStore?.workflowNode?.outputs[index].value.model.id.toString() ?? undefined
```
Comment (echl) [inline packages/client/hmi-client/src/page/project/tera-project.vue:244]: Thinking about how to make this more generic to support the other drilldowns... maybe `openedWorkflowNodeStore` should just store a `assetType` and `assetId`?

File: packages/client/hmi-client/src/page/project/components/tera-tab-content.vue [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+import CodeEditor from '@/page/project/components/code-editor.vue';
+import SimulationPlan from '@/page/project/components/Simulation.vue';
+import SimulationRun from '@/temp/SimulationResult3.vue';
+import TeraProjectOverview from '@/page/project/components/tera-project-overview.vue';
+import TeraSimulationWorkflow from '@/components/workflow/tera-simulation-workflow.vue';
+
+const props = defineProps<{
+	project: IProject;
+	assetId?: string;
+	assetType?: ProjectAssetTypes | 'overview' | 'workflow' | '';
```
Comment (echl) [inline packages/client/hmi-client/src/page/project/components/tera-tab-content.vue:85]: We might want to just add `overview` and `workflow`  and `` to `ProjectAssetTypes`

Comment (shawnyama) [reply packages/client/hmi-client/src/page/project/components/tera-tab-content.vue:85]: Workflow makes sense. But overview isn't an asset and if we were to add a process like calibrate it wouldn't make sense either. What do people think of having an `AssetType`, a `ProcessType` and a `ProjectPageType` that would hold the `AssetType` + `ProcessType` and any other project page like 'overview'.

**Review by echl — APPROVED** [review]
